discovery-provider/src/tasks/aggregates/index_aggregate_track.py

```diff
@@ -0,0 +1,138 @@
+
+logger = logging.getLogger(__name__)
+
+AGGREGATE_TRACK = "aggregate_track"
+
+# UPDATE_AGGREGATE_TRACK_QUERY
+# Get a lower bound blocknumber to check for new save/repost activity
+# Find a subset of tracks that have changed after that blocknumber
+# For that subset of tracks calculate the play/repost total counts
+# Insert that count for new users or update an existing row
+UPDATE_AGGREGATE_TRACK_QUERY = """
+    WITH new_track AS (
+        SELECT
+            t.track_id AS track_id
+        FROM
+            tracks t
+        GROUP BY
+            t.track_id
+        UNION
+        ALL (
+            SELECT
+                r.repost_item_id AS track_id
+            FROM
+                reposts r
+            WHERE
+                r.is_current IS TRUE
+                AND r.repost_type = 'track'
+                AND r.is_delete IS FALSE
+                AND r.blocknumber > :prev_blocknumber
+            GROUP BY
+                r.repost_item_id
+        )
+        UNION
+        ALL (
+            SELECT
+                s.save_item_id AS track_id
+            FROM
+                saves s
+            WHERE
+                s.is_current IS TRUE
+                AND s.save_type = 'track'
+                AND s.is_delete IS FALSE
+                AND s.blocknumber > :prev_blocknumber
+            GROUP BY
+                s.save_item_id
+        )
+    )
+    INSERT INTO
+        aggregate_track (track_id, repost_count, save_count)
+    SELECT
+        DISTINCT(t.track_id),
+        COALESCE(track_repost.repost_count, 0) AS repost_count,
+        COALESCE(track_save.save_count, 0) AS save_count
+    FROM
+        tracks t
+        LEFT OUTER JOIN (
+            SELECT
+                r.repost_item_id AS track_id,
+                count(r.repost_item_id) AS repost_count
+            FROM
+                reposts r
+            WHERE
+                r.is_current IS TRUE
+                AND r.repost_type = 'track'
+                AND r.is_delete IS FALSE
+                AND r.repost_item_id IN (
+                    SELECT
+                        track_id
+                    FROM
+                        new_track
+                )
+            GROUP BY
+                r.repost_item_id
+        ) AS track_repost ON track_repost.track_id = t.track_id
+        LEFT OUTER JOIN (
+            SELECT
+                s.save_item_id AS track_id,
+                count(s.save_item_id) AS save_count
+            FROM
+                saves s
+            WHERE
+                s.is_current IS TRUE
+                AND s.save_type = 'track'
+                AND s.is_delete IS FALSE
+                AND s.save_item_id IN (
+                    SELECT
+                        track_id
+                    FROM
+                        new_track
+                )
+            GROUP BY
+                s.save_item_id
+        ) AS track_save ON track_save.track_id = t.track_id
+    WHERE
+        t.track_id in (
```

Comment:
I have this unstaged change from our call:

```
        t.is_current is TRUE
        AND t.track_id in (
```

Reply:
Same for the above.

Reply:
Is `t.is_current is TRUE` unnecessary since we're already filtering for this when we create `new_track` referenced 4 lines below?

Reply:
Yeah because we don't want a bunch of duplicate rows before we join to track_repost and track_save.